zenoh-plugin-mqtt/src/lib.rs

```diff
@@ -544,12 +649,30 @@ async fn handshake_v5<'a>(
     handshake: v5::Handshake,
     zsession: Arc<Session>,
     config: Arc<Config>,
+    auth_dictionary: Arc<Option<HashMap<User, Password>>>,
 ) -> Result<v5::HandshakeAck<MqttSessionState<'a>>, MqttPluginError> {
     let client_id = handshake.packet().client_id.to_string();
-    log::info!("MQTT client {} connects using v5", client_id);
 
-    let session = MqttSessionState::new(client_id, zsession, config, handshake.sink().into());
-    Ok(handshake.ack(session))
+    match is_authorized(
+        (*auth_dictionary).as_ref(),
+        handshake.packet().username.as_ref(),
+        handshake.packet().password.as_ref(),
+    ) {
+        Ok(_) => {
+            log::info!("MQTT client {} connects using v5", client_id);
+            let session =
+                MqttSessionState::new(client_id, zsession, config, handshake.sink().into());
+            Ok(handshake.ack(session))
+        }
+        Err(err) => {
+            log::warn!(
+                "MQTT client {} connect using v5 rejected: {}",
+                client_id,
+                err
+            );
+            Ok(handshake.failed(ntex_mqtt::v5::codec::ConnectAckReason::BadUserNameOrPassword))
```

Comment:
`BadUserNameOrPassword` is for the case where the data in the user name or password is malformed.
See https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901075 table 3.2.2.2.
I think `NotAuthorized` is more appropriate here.

Reply:
Changed to `NotAuthorized`.